salt/states/idem.py

```diff
@@ -0,0 +1,149 @@
+# -*- coding: utf-8 -*-
+#
+# Author: Tyler Johnson <tjohnson@saltstack.com>
+#
+
+"""
+Idem Support
+============
+
+This state provides access to idem states
+"""
+from __future__ import absolute_import, print_function, unicode_literals
+
+import logging
+import os
+import sys
+
+import salt.modules.idem
+
+log = logging.getLogger(__name__)
+
+# Function alias to make sure not to shadow built-in's
+__func_alias__ = {"exec_": "exec"}
+__virtualname__ = "idem"
+
+
+def __virtual__():
+    if sys.version_info < (3, 6):
+        return False, "idem only works on python3.6 and later"
+    if not salt.modules.idem.HAS_POP[0]:
```

Comment:
Instead of cross-importing a state just to get it's Boolean value if it loaded, look for `idem.<a func name that should have loaded>` in `__salt__`

Reply:
It's not just the loaded - it also provides the correct error message, FWIW.

Reply:
Which the exec module would have logged, the message.

Reply:
Logged, or presented to the user?

Reply:
Salt doesn't present anything to the user when a module does not load. Or state module. It logs and wines that the module could not be found